lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -53,6 +56,192 @@ class QcFlagService {
         return qcFlag;
     }
 
+    /**
+     * Create new instance of quality control flags for data pass
+     * @param {Partial<QcFlag>} parameters flag instance parameters
+     * @param {object} [relations] QC Flag Type entity relations
+     * @param {Partial<UserIdentifier>} [relations.user] user identifiers
+     * @param {number} [parameters.flagTypeId] flag type id
+     * @param {number} [parameters.runNumber] associated run's number
+     * @param {number} [parameters.dataPassId] associated dataPass' id
+     * @param {number} [parameters.dplDetectorId] associated dplDetector's id
+     * @return {Promise<QcFlag>} promise
+     * @throws {BadParameterError, NotFoundError}
+     */
+    async createForDataPass(parameters, relations = {}) {
+        const {
+            from: fromTime = null,
+            to: toTime = null,
+            comment,
+        } = parameters;
+        const {
+            user: { userId, externalUserId } = {},
+            flagTypeId,
+            runNumber,
+            dataPassId,
+            dplDetectorId,
+        } = relations;
+
+        if (fromTime && toTime && !(fromTime < toTime)) {
```

Comment:
Prefer fromTime >= toTime, go to the simplest

Reply:
applied